lite/examples/sound_classification/android_compose/app/src/main/java/org/tensorflow/lite/examples/soundclassifier/compose/MainActivity.kt

```diff
@@ -0,0 +1,164 @@
+package org.tensorflow.lite.examples.soundclassifier.compose
+
+import android.Manifest
+import android.content.pm.PackageManager
+import android.media.AudioRecord
+import android.os.Build
+import android.os.Bundle
+import android.os.Handler
+import android.os.HandlerThread
+import android.util.Log
+import android.view.WindowManager
+import androidx.activity.ComponentActivity
+import androidx.activity.compose.setContent
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.activity.viewModels
+import androidx.annotation.RequiresApi
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.core.content.ContextCompat
+import androidx.core.os.HandlerCompat
+import androidx.lifecycle.lifecycleScope
+import kotlinx.coroutines.flow.collect
+import org.tensorflow.lite.examples.soundclassifier.compose.ui.SoundClassifierScene
+import org.tensorflow.lite.examples.soundclassifier.compose.ui.SoundClassifierViewModel
+import org.tensorflow.lite.task.audio.classifier.AudioClassifier
+
+class MainActivity : ComponentActivity() {
```

Comment:
Most of the business logic if not all should sit in the viewmodel instead of the activity

Reply:
Done!